File: services/domain-api/src/main/java/com/example/iter/reservation/api/RentalScheduleItem.java
package com.example.iter.reservation.api;

import java.time.LocalDate;
import java.util.Objects;

// 장비의 예약 일정 한 칸. device 의 일정 조회 화면이 유일한 소비자다.
public record RentalScheduleItem(
        Long rentalId,
        LocalDate startDate,
        LocalDate endDate,
        RentalStatus status
) {
    public RentalScheduleItem {
        Objects.requireNonNull(rentalId, "rentalId");
        Objects.requireNonNull(status, "status");
    }
}

File: services/domain-api/src/main/kotlin/com/example/iter/reservation/api/RentalScheduleItem.kt
package com.example.iter.reservation.api

import java.time.LocalDate

// 장비의 예약 일정 한 칸. device 의 일정 조회 화면이 유일한 소비자다.
@JvmRecord
data class RentalScheduleItem(
    val rentalId: Long,
    val startDate: LocalDate,
    val endDate: LocalDate,
    val status: RentalStatus,
)
